sphinx/directives/other.py

```diff
@@ -118,7 +119,7 @@ def parse_content(self, toctree: addnodes.toctree) -> List[Node]:
                 docname = docname_join(self.env.docname, docname)
                 if url_re.match(ref) or ref == 'self':
                     toctree['entries'].append((title, ref))
-                elif docname not in self.env.found_docs:
+                elif docname not in (self.env.found_docs | generated_documents):
```

Comment:
It would be better to move this union computation out of the loop here

Reply:
Fixed